solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -179,34 +262,78 @@ public String toString(String field) {
     return field;
   }
 
+  public class FeatureInfo {
+      String name;
+      float value;
+      boolean used;
+      
+      FeatureInfo(String n, float v, boolean u){
+        name = n; value = v; used = u;
+      }
+       
+      public boolean getUsed(){
+        return used;
+      }
+      
+      public void setUsed(boolean u){
+        used = u;
+      }
+      
+      public void setScoreUsed(float score, boolean used){
+        this.value = score;
+        this.used = used;
+      }
+      
+      public String getName(){
+        return name;
+      }
+      
+      public float getValue(){
+        return value;
+      }
+  }
+  
   public class ModelWeight extends Weight {
 
     IndexSearcher searcher;
 
     // List of the model's features used for scoring. This is a subset of the
     // features used for logging.
-    FeatureWeight[] modelFeatures;
+    FeatureWeight[] modelFeatureWeights;
     float[] modelFeatureValuesNormalized;
+    FeatureWeight[] extractedFeatureWeights;
+    Normalizer[] modelFeatureNorms;
 
-    // List of all the feature values, used for both scoring and logging
-    FeatureWeight[] allFeatureWeights;
-    float[] allFeatureValues;
-    String[] allFeatureNames;
-    boolean[] allFeaturesUsed;
+    // List of all the feature names, values, used for both scoring and logging
+    LinkedHashMap<Integer, FeatureInfo> featuresInfo;
 
-    public ModelWeight(IndexSearcher searcher, FeatureWeight[] modelFeatures,
-        FeatureWeight[] allFeatures) {
+    public ModelWeight(IndexSearcher searcher, FeatureWeight[] modelFeatureWeights,
```

Comment:
Not needed but left there for backwards compatibility. On second thought, its ok to remove as no one else is using this code yet